src/tsc-errors/error-store.ts

```diff
@@ -0,0 +1,57 @@
+import { TscError, TscErrorMatchFlags } from './types';
+
+class ErrorStore {
+  private tscErrors: Map<TscErrorMatchFlags, TscError[]> = new Map();
+
+  public getAllErrors(): TscError[] {
+    return [...this.tscErrors.values()].flat();
+  }
+
+  public getIgnoredErrors(): TscError[] {
+    const key = TscErrorMatchFlags.errorCode | TscErrorMatchFlags.filePath;
+    return this.tscErrors.get(key) || [];
+  }
+
+  public getReportedErrors(): TscError[] {
+    const keys = [
+      TscErrorMatchFlags.errorCode & TscErrorMatchFlags.filePath,
+      TscErrorMatchFlags.errorCode,
+      TscErrorMatchFlags.filePath,
+    ];
+    const errors: TscError[] = [];
+    for (let key of keys) {
+      const values = this.tscErrors.get(key) || [];
+      errors.push(...values);
+    }
+    return errors;
+  }
+
+  public getValidErrors(): TscError[] {
```

Comment:
Minor:

All errors are valid, but some are ignored, and some are not ignored. Can we use the name

```suggestion
  public getUnignoredErrors(): TscError[] {
```

or use some other word to avoid the confusion around calling them _valid_ errors :sweat_smile: 

I know this probably was the name used in my initial implementation, but this occurred to me as I was reading this code. Feel free to skip this suggestion